build/gh-pages-app.js

```diff
@@ -142,19 +145,30 @@ class GhPagesApp {
 
   /**
    * @param {string} input
+   * @param {rollup.Plugin[]=} plugins
    * @return {Promise<string>}
    */
-  async _rollupSource(input) {
-    const plugins = [
-      nodeResolve(),
-      commonjs(),
+  async _rollupSource(input, plugins) {
+    plugins = plugins || [
+      rollupPlugins.nodeResolve(),
+      rollupPlugins.commonjs(),
     ];
-    if (!process.env.DEBUG) plugins.push(rollupTerser());
+    if (!process.env.DEBUG) plugins.push(rollupPlugins.terser());
+
     const bundle = await rollup.rollup({
       input,
       plugins,
     });
-    const {output} = await bundle.generate({format: 'iife'});
+    const {output} = await bundle.generate({format: 'esm'});
```

Comment:
Changed this because rollup won't do chunks for iife. You want to see a separate PR that only changes the output format to `esm`?

Reply:
> You want to see a separate PR that only changes the output format to esm?

No I was asking what impact this has, and to see a PR that separated all of these rollup structure changes from the locale swap changes if that impact was large.

Going from 

```js
(() => {
function foo() {

}

foo()
})()
```

to 
```
import {foo} from './chunk'
foo()
```

(is that actually what happens?) across all our bundles feels significant to happen in a PR labeled only viewer changes

Reply:
> all our bundles

It's only for the gh-pages-app bundles (viewer and treemap). As long as the entry file has no exports, and no externals are defined (which would result in toplevel imports; note: externals would break us regardless of format), there will be no toplevel imports/export statements.

Only the viewer include `swap-locale-features`, which means only the viewer bundle will use a _dynamic_ import (which is safe to use even outside a module ie. a common `<script>`).

Ideally this `gh-pages-app` would always be iife, as we concact these various scripts together and so the general case of concating multiple esm modules isn't guaranteed to work... Perhaps GhPagesApp should be tweaked to emit a separate `script` resource for every rollup entry? But then we have issues of dependency requirements, which is complex across `<script module>` since they load async...

I don't anticipate any issues with this, and IIRC this is an isolated change (nothing else needed to change), so extracting to a new PR won't make this one any simpler to review. That said, happy to do a quick PR for it for a cleaner diff/commit message.

Reply:
Mmk, I'll do another pass then.